aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+                rpc_port: dora_node_info.rpc_port,
+            };
+            vector::push_back(&mut node_data_vec, node_data);
+        };
+        node_data_vec
+    }
+
+    #[view]
+    /// Get the committee's ids from the store
+    public fun get_committee_ids(com_store_addr: address): vector<u64> acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        simple_map::keys(&committee_store.committee_map)
+    }
+
+    #[view]
+    /// Get the committee's id for a single node, only pass the address is okay
+    public fun get_committee_id(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        *simple_map::borrow(&committee_store.node_to_committee_map, &node_address)
+    }
+
+    #[view]
+    /// Get the node's information
+    public fun get_node_info(
+        com_store_addr: address,
+        id: u64,
+        node_address: address
+    ): NodeData acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow(&committee_store.committee_map, &id);
+        let (addrs, dora_nodes) = simple_map::to_vec_pair(committee.map);
+        let (flag, index) = vector::index_of(&addrs, &node_address);
+        assert!(flag, error::invalid_argument(NODE_NOT_FOUND));
+        let dora_node_info = vector::borrow(&dora_nodes, index);
+
+        NodeData {
+            operator: node_address,
+            ip_public_address: dora_node_info.ip_public_address,
+            dora_public_key: dora_node_info.dora_public_key,
+            network_public_key: dora_node_info.network_public_key,
+            elgamal_pub_key: dora_node_info.elgamal_pub_key,
+            network_port: dora_node_info.network_port,
+            rpc_port: dora_node_info.rpc_port,
+        }
+    }
+
+    #[view]
+    /// Get the committee's id for a single node
+    public fun get_committee_id_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): u64 acquires CommitteeInfoStore {
+        let committee_store = borrow_global<CommitteeInfoStore>(com_store_addr);
+        let id = *simple_map::borrow(&committee_store.node_to_committee_map, &node_address);
+        id
+    }
+
+    #[view]
+    /// Get the dora node peers vector for a single node
+    public fun get_peers_for_node(
+        com_store_addr: address,
+        node_address: address
+    ): vector<NodeData> acquires CommitteeInfoStore {
+        let committee_id = get_committee_id_for_node(com_store_addr, node_address);
+        let this_node = get_node_info(com_store_addr, committee_id, node_address);
+        let dora_node_info = get_committee_info(com_store_addr, committee_id);
+        let (_, index) = vector::index_of(&dora_node_info, &this_node);
+        vector::remove(&mut dora_node_info, index);
```

Comment:
Why did we remove the requested node_address details from the NodeData list, @axiongsupra?

Reply:
@nizam-supraoracles Let's consider we are trying to get peers for A, which might be B, C, ... However, A itself is not consider as peer's for A.

Reply:
In that case, we need one more function which list out itself and there peer list

Reply:
Hi @nizam-supraoracles , given a `committee_id` one can query entire list of nodes part of that committee. If a node is querying for `peers`, wouldn't it be fair to assume that it knows about itself (parameters such as various keys/ports)? In which context are you planning to use this? It would help us better understand the need. Also, this is a stale PR, use #7 for commenting/reviewing further.

Reply:
In that case, we need to call two different view functions. The first is to get `get_committee_id`, and then use the committee_id as a parameter to call the `get_committee_info` function.

Sure I will use https://github.com/Entropy-Foundation/aptos-core/pull/7 PR for the next upcoming review.
Since this PR is now stale, we can close it.